Fail to build with Qt 5.9 beta 2

<!--- Provide a general summary of the issue in the title above -->

## Expected Behavior
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->
KeepassXC builds fine even with the latest beta preview of Qt

## Current Behavior
<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from the current behavior -->
```
[ 73%] Linking CXX executable testentrymodel
cd /home/yen/Projects/tmp/keepassxc/tests && /usr/bin/cmake -E cmake_link_script CMakeFiles/testentrymodel.dir/link.txt --verbose=1
/usr/bin/c++   -fno-common -Wall -Wextra -Wundef -Wpointer-arith -Wno-long-long -Wformat=2 -Wmissing-format-attribute -fvisibility=hidden -fvisibility-inlines-hidden -fstack-protector-strong -fno-exceptions -fno-rtti -Wnon-virtual-dtor -Wold-style-cast -Woverloaded-virtual -D_FORTIFY_SOURCE=2 -Werror=format-security -Wcast-align -pie -fPIE -std=c++11 -O2 -g -DNDEBUG   -Wl,--no-add-needed -Wl,--as-needed -Wl,--no-undefined -Wl,-z,relro,-z,now -rdynamic CMakeFiles/testentrymodel.dir/TestEntryModel.cpp.o CMakeFiles/testentrymodel.dir/testentrymodel_autogen/moc_compilation.cpp.o  -o testentrymodel libtestsupport.a ../src/libkeepassx_core.a /usr/lib/libQt5Concurrent.so.5.9.0 /usr/lib/libQt5Test.so.5.9.0 -lgcrypt -lgpg-error -lz ../src/libautotype.a /usr/lib/libQt5Widgets.so.5.9.0 /usr/lib/libQt5Gui.so.5.9.0 ../src/libzxcvbn.a /usr/lib/libQt5DBus.so.5.9.0 /usr/lib/libQt5Core.so.5.9.0
libtestsupport.a(modeltest.cpp.o): In function `ModelTest::data()':
/home/yen/Projects/tmp/keepassxc/tests/modeltest.cpp:451: undefined reference to `bool QTest::qCompare<int, unsigned int>(int const&, unsigned int const&, char const*, char const*, char const*, int)'
collect2: error: ld returned 1 exit status
make[2]: *** [tests/CMakeFiles/testentrymodel.dir/build.make:134: tests/testentrymodel] Error 1
make[2]: Leaving directory '/home/yen/Projects/tmp/keepassxc'
make[1]: *** [CMakeFiles/Makefile2:1175: tests/CMakeFiles/testentrymodel.dir/all] Error 2
make[1]: Leaving directory '/home/yen/Projects/tmp/keepassxc'
make: *** [Makefile:141: all] Error 2
```
## Possible Solution
<!--- Not obligatory, but suggest a fix/reason for the bug, -->
<!--- or ideas how to implement the addition or change -->

Inspired by http://stackoverflow.com/a/14198973/3786245, I force the second to be int:
```Diff
diff --git a/tests/modeltest.cpp b/tests/modeltest.cpp
index 360a7be..0792e4f 100644
--- a/tests/modeltest.cpp
+++ b/tests/modeltest.cpp
@@ -448,7 +448,7 @@ void ModelTest::data()
     QVariant textAlignmentVariant = model->data ( model->index ( 0, 0 ), Qt::TextAlignmentRole );
     if ( textAlignmentVariant.isValid() ) {
         int alignment = textAlignmentVariant.toInt();
-        QCOMPARE( alignment, ( alignment & ( Qt::AlignHorizontal_Mask | Qt::AlignVertical_Mask ) ) );
+        QCOMPARE( alignment, static_cast<int>( alignment & ( Qt::AlignHorizontal_Mask | Qt::AlignVertical_Mask ) ) );
     }
 
     // General Purpose roles that should return a QColor
``` 

## Steps to Reproduce (for bugs)
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
1. git clone keepassxc
2. cmake .
3. make

## Context
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->

## Your Environment
<!--- Include relevant details about the environment you experienced the bug in -->
* KeePassXC version/commit used: (can be found under Help -> About) ad79162061c19decdd122784ff8adc4f18d6ce9b
* Qt version (e.g. Qt 5.3): Qt 5.9 beta 2
* Compiler (e.g. Clang++3.6.0): GCC 6.3.1
* Operating System and version: Arch Linux x86_64 (with [kde-unstable] repository enabled)

--- a/tests/modeltest.cpp
+++ b/tests/modeltest.cpp
@@ -448,7 +448,8 @@ void ModelTest::data()
     QVariant textAlignmentVariant = model->data ( model->index ( 0, 0 ), Qt::TextAlignmentRole );
     if ( textAlignmentVariant.isValid() ) {
         int alignment = textAlignmentVariant.toInt();
-        QCOMPARE( alignment, ( alignment & ( Qt::AlignHorizontal_Mask | Qt::AlignVertical_Mask ) ) );
+        QCOMPARE( alignment, static_cast<int>( alignment & ( Qt::AlignHorizontal_Mask
+                                                           | Qt::AlignVertical_Mask ) ) );
     }
 
     // General Purpose roles that should return a QColor

